southpark-search/tests/test_southparksearch.py

```diff
@@ -49,46 +49,41 @@ def index_documents():
 
 
 def call(method, url, payload=None, headers={'Content-Type': 'application/json'}):
-	import requests
-	return getattr(getattr(requests, method)(url, data=json.dumps(payload), headers=headers), 'json')()
+    import requests
+    return getattr(getattr(requests, method)(url, data=json.dumps(payload), headers=headers), 'json')()
 
 
 def get_results(query, top_k=top_k):
     return call('post', 
-    	        'http://0.0.0.0:45678/api/search', 
-    	        payload={"top_k": top_k, "mode": "search",  "data": [f"text:{query}"]}
-    	        )
+                'http://0.0.0.0:45678/api/search', 
+                payload={"top_k": top_k, "mode": "search",  "data": [f"text:{query}"]}
+                )
 
 
 def set_flow():
-	f = Flow().load_config(query_flow_file_path)
-	f.use_rest_gateway()
-	return f
+    f = Flow().load_config(query_flow_file_path)
+    f.use_rest_gateway()
+    return f
 
 
 @pytest.fixture
 def queries():
-	return [('hey dude', ["Don't say anything\n", "Check that: I'll watch that game.\n", "Because you're a fucking fatass\n"]),
-	        ('sister', ['Sorry\n', 'Lame.\n', 'Christ\n']),
-	        ("Ill watch that game", ["Check that: I'll watch that game.\n", 'Quit it\n', 'Sorry\n'])]
-
-
-def test_query(queries):
-	config()
-	setup_env()
-	prepare_data()
-	index_documents()
-	f = set_flow()
-	with f:
-		for query, exp_result in queries:
-			output = get_results(query)
-			matches = output['search']['docs'][0]['matches']
-			assert len(matches) == top_k #check the number of docs returned
-			result = []
-			for match in matches:
-				result.append(match['text'])
-			assert result == exp_result
-
-
-def test_cleanup():
-	shutil.rmtree(os.environ['JINA_WORKSPACE'])
+    return [('hey dude', ["Don't say anything\n", "Check that: I'll watch that game.\n", "Because you're a fucking fatass\n"]),
+            ('sister', ['Sorry\n', 'Lame.\n', 'Christ\n']),
+            ("Ill watch that game", ["Check that: I'll watch that game.\n", 'Quit it\n', 'Sorry\n'])]
+
+
+def test_query(queries, tmpdir):
+    config(tmpdir)
+    prepare_data()
```

Comment:
I do not understand why do we do `prepare_data` if we are passing it?

Reply:
I wanted to keep the whole code as a template to be used for other examples. Initially, I was preparing the actual original data in Southpark and then use it but now that we are making custom small data for all examples, I will remove these.

Reply:
Makes sense, good idea, just maybe we should add a comment there to guide potential future devs?

Reply:
I have removed it for now as we use manually made small datasets. So we don't need prepare_data anymore